### ISSUE TITLE
ensureSafeRefName warnings are emitted on stubbed nodes

### ISSUE THREAD

**Issue by wbecker** [QubitProducts/react-test-tree#39]
In the following case a warning is emitted, however it probably shouldn't because the component that is causing the warning is stubbed out. When you are using a 3rd party library that you are stubbing out, you don't want to be warned about their problems!

```
var BizComponent = React.createClass({
  render: function () {
    return (
      <div>
        <button ref="value">Fuz</button>
      </div>
    );
  }
});

var FooComponent = React.createClass({
  render: function () {
    return (
      <div>
        <BizComponent ref="biz"/>
      </div>
    );
  }
});

var fooTree = testTree(<FooComponent />, {
  stub: {
    biz: {}
  }
});
```

**Comment by oliverwoodings** [comment]
You have no idea how excited I got when I got an email from 'William Becker' pop up in my inbox

![](https://media1.giphy.com/media/mpoYCxf1BUU5W/200.gif)

I'm not sure why you are getting that warning - it shouldn't even be creating the node, let alone reading it's refs! I'll look into it when I get some time.

**Comment by wbecker** [comment]
ZOMG! Thanks for that, works well otherwise.

Might see you Friday!

**Comment by oliverwoodings** [comment]
Friday?

**Comment by roberttod** [comment]
![movie on 12-08-2015 at 15 44](https://cloud.githubusercontent.com/assets/823104/9226973/1d321664-4109-11e5-950b-a6c147a1f595.gif)
My reaction ^

**Comment by oliverwoodings** [comment]
![](http://g.recordit.co/QtGzlZIsiC.gif)

My reaction every morning when I arrive at work ^

**Comment by wbecker** [comment]
:blush:

**Comment by oliverwoodings** [comment]
Closed by #31
